### PR TITLE
Support remote rg lz referencing

### PR DESCRIPTION
**Pull request by seanlok**
Current Storage account only can use the RG that is created within the same tfstate only. Adding support for remote lz_key and key in order to be able to deploy to remote RG if required.

### CODE REVIEW

File: storage_accounts.tf [hunk 1]
```diff
@@ -6,13 +6,13 @@ module "storage_accounts" {
   global_settings     = local.global_settings
   client_config       = local.client_config
   storage_account     = each.value
-  resource_group_name = local.resource_groups[each.value.resource_group_key].name
-  location            = lookup(each.value, "region", null) == null ? local.resource_groups[each.value.resource_group_key].location : local.global_settings.regions[each.value.region]
+  resource_group_name = local.combined_objects_resource_groups[try(each.value.resource_group.lz_key, local.client_config.landingzone_key)][try(each.value.resource_group.key, each.value.resource_group_key)].name
```
Comment (LaurentLesle) [inline storage_accounts.tf:9]: can you refactor with a coalesce to make it more readable

File: storage_accounts.tf [hunk 2]
```diff
@@ -6,13 +6,13 @@ module "storage_accounts" {
...
+  location            = lookup(each.value, "region", null) == null ? local.combined_objects_resource_groups[try(each.value.resource_group.lz_key, local.client_config.landingzone_key)][try(each.value.resource_group.key, each.value.resource_group_key)].location : local.global_settings.regions[each.value.region]
```
Comment (LaurentLesle) [inline storage_accounts.tf:10]: same comment here with coalesce

File: storage_accounts.tf [hunk 3]
```diff
@@ -6,13 +6,13 @@ module "storage_accounts" {
...
   vnets               = local.combined_objects_networking
   private_endpoints   = try(each.value.private_endpoints, {})
   resource_groups     = try(each.value.private_endpoints, {}) == {} ? null : local.resource_groups
   recovery_vaults     = local.combined_objects_recovery_vaults
-  base_tags           = try(local.global_settings.inherit_tags, false) ? local.resource_groups[each.value.resource_group_key].tags : {}
+  base_tags           = try(local.global_settings.inherit_tags, false) ? local.combined_objects_resource_groups[try(each.value.resource_group.lz_key, local.client_config.landingzone_key)][try(each.value.resource_group.key, each.value.resource_group_key)].tags : {}
```
Comment (LaurentLesle) [inline storage_accounts.tf:15]: and here

**Review by LaurentLesle — CHANGES_REQUESTED** [review]

**Review by arnaudlh — CHANGES_REQUESTED** [review]
hi @seanlok, couple of things:
- coalese or coalesce? :) 
- couple of syntax errors
can you please review, test locally and update? thanks

**Comment by seanlok** [comment]
This s fixed and tested.

**Review by arnaudlh — APPROVED** [review]

**Comment by arnaudlh** [comment]
@seanlok - thank you for your contribution to Cloud Adoption Framework for Azure!

**Pull request merged**